### PR TITLE
Autosar SOME/IP protocol support

### PR DESCRIPTION
**Pull request by ffatghub**
Hi, this is basic support for the Autosar SOME/IP protocol (Scalable service-Oriented MiddlewarE over IP) dissection. I am new in the automotive field, I needed a way to track down a problem in an embedded system and tcpdump printing someip packet details helped me a lot, grep-ing for service/client/session in particular. I hope I did all necessary stuff, let me know... I can work on this next week.
Ciao thank you
francesco

### CODE REVIEW

**Comment by guyharris** [comment]
Use the standard C99  uintN_t types, not the u_intN_t types; that's why the build is failing on Windows.

**Comment by ffatghub** [comment]
> 
> 
> Use the standard C99 uintN_t types, not the u_intN_t types; that's why the build is failing on Windows.

Hi Guy, I see, thanks. Done in cb9ed6c.

File: print-someip.c [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+    message_len = GET_BE_U_4(bp);
+    bp += 4;
+
+    request_id = GET_BE_U_4(bp);
+    client_id = request_id >> 16;
+    session_id = request_id & 0x0000FFFF;
+    bp += 4;
+
+    protocol_version = *bp;
```
Comment (guyharris) [inline print-someip.c]: Please use `GET_U_1()` when fetching single-byte unsigned integer values.  That way, it does bounds checking on the packet data.

Comment (ffatghub) [reply print-someip.c]: Done in 246ca11. Thanks. Ciao

**Pull request merged**